gatsby-node.js

```diff
@@ -63,29 +70,36 @@ exports.sourceNodes = async (gatsbyOptions, pluginOptions) => {
 	// Create a node for each image and store it in the cache
 	// so it can bre retrieved on file resolution.
 	const remoteImages = await plugin.getAllImages();
-	remoteImages.forEach(async (image) => {
-		const cachedImage = await cache.get(image.id)
-		if (cachedImage !== undefined) {
-			return
-		}
-		const nameParts = image.filename_download.split('.');
-		const ext = nameParts.length > 1 ? `.${nameParts.pop()}` : '';
-		const name = nameParts.join('.');
-		const imageUrl = `${plugin.url}assets/${image.id}`;
-		const img = await createRemoteFileNode({
-			url: imageUrl,
-			parentNodeId: image.id,
-			store,
-			cache,
-			createNode,
-			createNodeId,
-			httpHeaders: { Authorization },
-			reporter,
-			ext,
-			name,
-		});
-		await cache.set(image.id, img);
-	})
+
+	await Promise.all(
```

Comment:
Lets see if we can do this in a batched approach. Downloading all files at the same time might work for smaller installations, but you'll most likely run into rate limit / download speed limits if Promise.all-ing hundreds/thousands of files

Reply:
Fixed it  👍

Reply:
@rijkvanzanten Actually gatsby-source-filesystem has it‘s own batching logic though a variable („GATSBY_CONCURRENT_DOWNLOAD“). So I am not too sure how valuable an additional batching is but it probably won’t hurt.